### PR TITLE
Add queen-attack exercise

### PR DESCRIPTION
**Pull request by BNAndras**
Related to #172

### CODE REVIEW

**Comment by BNAndras** [comment]
I deviated from the auto-generated test suite which only used the Create function for the first few tests and then passed dictionaries representing positions into CanAttack without going through Create. That didn't seem that interesting so I made Create return a dictionary representing a Queen with a stored CanAttack function.

Since the first test requires a dictionary be returned, students should hopefully make the connection to storing functions in it. However, this might require an instructions append outlining that Vim script doesn't really have classes so that's why we're using a dictionary and maybe a link about Vim script dictionaries.

File: exercises/practice/queen-attack/.meta/example.vim [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    endif
+
+    return {'row': a:row, 'column': a:column, 'CanAttack': function('DoCanAttack')}
+endfunction
+
+function! DoCanAttack(other) dict
+    return self.row == a:other.row ||
+    \ self.column == a:other.column ||
+    \ (abs(a:other.column - self.column) == abs(a:other.row - self.row))
+endfunction
```
Comment (kotp) [inline exercises/practice/queen-attack/.meta/example.vim:19]: Missing the last line (as a terminator, not a separator).

**Review by kotp — APPROVED** [review]
Hopefully the missing EOL character will come in before this is merged, but pre-approving.

File: exercises/practice/queen-attack/.meta/example.vim (hunk 1 above)
Comment (kotp) [inline exercises/practice/queen-attack/.meta/example.vim:19]:
```suggestion
endfunction

```

Comment (BNAndras) [reply exercises/practice/queen-attack/.meta/example.vim:19]: Oops, I didn’t catch that. Thanks!

**Pull request merged**